### PR TITLE
Some diagrams for BoxFit.

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
@HansMuller

File: painting/box_fit.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+
+void main() {
+  List<Widget> children = <Widget>[];
```
Comment (HansMuller) [inline painting/box_fit.dart]: Our linter would have crabbed about this not being final :-)

**Comment by HansMuller** [comment]
This is pretty slick. All those diagrams for just a little code. NICE.

LGTM

File: painting/box_fit.dart (hunk 1 above)
Comment (Hixie) [inline painting/box_fit.dart]: la la la? :-)

**Pull request merged**